src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs

```diff
@@ -168,8 +168,16 @@ private JObject InitializeHostConfig(string hostJsonPath, JObject hostConfigObje
                         TryWriteHostJson(hostJsonPath, hostConfigObject);
                     }
 
-                    if (hostConfigObject["version"]?.Value<string>() != "2.0")
+                    string hostJsonVersion = hostConfigObject["version"]?.Value<string>();
+                    if (hostJsonVersion != "2.0")
```

Comment:
this would be true if hostJsonVersion is set to v4.0

Reply:
Yeah so I think the logic is --
- if it's 2.0, we're good
- if it's not..
-- if it's 3.0, throw special exception
-- if it's anything else (like 4.0), throw generic exception

Reply:
- If its 4.0 right now the logic says the version is not present. 
- For 3.0 we have a specific error message but if customer sets it to 3 we are lost again. We may want to change and use the version property in the host.json to say it is invalid. That way it cover 3.0 and other strings that are not 2.0.

Reply:
nit: `string.Equals`